### ISSUE TITLE
Preprocessing CLI (and maybe all CLIs) can't handle spaces in the filepaths

### ISSUE THREAD

**Issue by dboun** [CBICA/CaPTk#860]
**Describe the bug**
```bash
###
### Escaping space doesn't work
###
dimitris@dimitris-lab:~/Documents/Programs/CaPTk-1.7.6-stable/CaPTk/1.7.6$ ./captk Preprocessing -i /home/dimitris/Documents/Medical\ Datasets/brain_tumor_5m/flair_5m.nii.gz -o /home/dimitris/Desktop/outdelll.nii.gz -zn
terminate called after throwing an instance of 'itk::ExceptionObject'
  what():  /home/vsts/work/1/s/bin/ITK-source/Modules/Remote/MGHIO/src/itkMGHImageIO.cxx:160:
itk::ERROR: MGHImageIO(0x1f4d7f0): A FileName must be specified.
Aborted (core dumped)
### 
### Quotation marks around the path with the space doesn't work
###
dimitris@dimitris-lab:~/Documents/Programs/CaPTk-1.7.6-stable/CaPTk/1.7.6$ ./captk Preprocessing -i "/home/dimitris/Documents/Medical Datasets/brain_tumor_5m/flair_5m.nii.gz" -o /home/dimitris/Desktop/outdelll.nii.gz -zn
terminate called after throwing an instance of 'itk::ExceptionObject'
  what():  /home/vsts/work/1/s/bin/ITK-source/Modules/Remote/MGHIO/src/itkMGHImageIO.cxx:160:
itk::ERROR: MGHImageIO(0x1ce77f0): A FileName must be specified.
Aborted (core dumped)
###
### Moving the image to a path without spaces works!
###
dimitris@dimitris-lab:~/Documents/Programs/CaPTk-1.7.6-stable/CaPTk/1.7.6$ ./captk Preprocessing -i /home/dimitris/Desktop/flair_5m.nii.gz -o /home/dimitris/Desktop/outdelll.nii.gz -zn
Cut-offs calculated: Lower = -26.8689; Upper = 379.75
dimitris@dimitris-lab:~/Documents/Programs/CaPTk-1.7.6-stable/CaPTk/1.7.6$
```

**CaPTk Version**
1.7.6

**Desktop (please complete the following information):**
 - OS: Linux Mint 19.3

**Additional context**
- Also tried it for n3 and ss and can confirm that the issue is there for everything.
- This might be an issue on other CLIs too?
- Other seemingly different issues might be related to this, like #854 #858 (marked as critical because of this)

**Comment by AlexanderGetka-cbica** [comment]
I investigated this and believe I have found an explanation and solution.
The shell is responsible for tokenizing the arguments _given to CaPTk_ when appropriate -- such as if the user puts quotes around their directory containing spaces.
Under normal circumstances this would be acceptable, but since we then directly pass those args via a string to the std::system() call for the CWL application, they reach the CWL executable in raw form (i.e., not properly tokenized). 
On Ubuntu 18.04, I was able to fix this issue on the same test cases by modifying line 112 of CaPTk.cxx to add quotes around every argument added to the "commandToRun" string which gets passed to std::system(). 
However, because std::system() is (obviously) platform dependent, this will definitely need testing on Windows and Mac.
